dns-policy-operator/src/dns_policy.py

```diff
@@ -0,0 +1,339 @@
+        """
+        try:
+            # Installing the charm via subprocess.
+            # Calling subprocess here is not a security issue.
+            logger.info(
+                "Installing from custom dns-policy snap located: %s",
+                snap_path,
+            )
+            subprocess.check_output(["sudo", "snap", "install", snap_path, "--dangerous"])  # nosec
+        except (snap.SnapError, snap.SnapNotFoundError) as e:
+            error_msg = f"An exception occurred when installing {snap_path}. Reason: {e}"
+            logger.exception(error_msg)
+            raise InstallError(error_msg) from e
+        except subprocess.CalledProcessError as e:
+            error_msg = f"An exception occurred when installing {snap_path}. Reason: {e}. Output: {e.output}"
+            logger.exception(error_msg)
+            raise InstallError(error_msg) from e
+
+    def get_config(self) -> dict:
+        """Get the configuration of the dns-policy service.
+
+        Returns:
+            dict of configuration values
+
+        Raises:
+            ConfigureError: when encountering a SnapError
+        """
+        try:
+            cache = snap.SnapCache()
+            charmed_bind = cache[constants.DNS_SNAP_NAME]
+            return charmed_bind.get(None, typed=True)
+        except snap.SnapError as e:
+            error_msg = (
+                f"An exception occurred when configuring {constants.DNS_SNAP_NAME}. Reason: {e}"
+            )
+            logger.error(error_msg)
+            raise ConfigureError(error_msg) from e
+
+    def configure(self, config: dict[str, str]) -> None:
+        """Configure the dns-policy service.
+
+        Args:
+            config: dict of configuration values
+
+        Raises:
+            ConfigureError: when encountering a SnapError
+        """
+        try:
+            cache = snap.SnapCache()
+            charmed_bind = cache[constants.DNS_SNAP_NAME]
+            charmed_bind.set(config)
+        except snap.SnapError as e:
+            error_msg = (
+                f"An exception occurred when configuring {constants.DNS_SNAP_NAME}. Reason: {e}"
+            )
+            logger.error(error_msg)
+            raise ConfigureError(error_msg) from e
+
+    def command(self, cmd: str) -> str:
+        """Run manage command of the dns-policy service.
+
+        Args:
+            cmd: command to execute by django's manage script
+            env: environment
+
+        Returns:
+            The resulting output of the command's execution
+        """
+        try:
+            # We ignore security issues with this subprocess call
+            # as it can only be done from the operator of the charm
+            return subprocess.check_output(
+                ["sudo", "snap", "run", f"{constants.DNS_SNAP_NAME}.manage"] + cmd.split(),
+            ).decode(  # nosec
+                "utf-8"
+            )
+        except subprocess.SubprocessError as e:
+            return f"Error: {e}"
```

Comment:
No really great, should throw a custom error

Reply:
Fixed